### PR TITLE
Make clicking title in docs behave like toggle button

### LINKED ISSUE #7975: [www] Inconsistent accordion behavior in docs sidebar
The title and button in sidebar sections behave differently when clicked. The title will collapse all other sections while the button keeps surrounding sections open. This causes inconsistency and in the case of title, some jumping of the sidebar. @KyleAMathews and I discussed this and have a fix I can implement.

### PR DESCRIPTION
**Pull request by kkemple**
This PR makes the behavior between clicking a title and the accompanying button the same.

Closes #7975

### CODE REVIEW

File: www/src/components/sidebar/sidebar.js [hunk 1]
```diff
@@ -235,7 +235,7 @@ class SidebarBody extends Component {
                 key={index}
                 level={0}
                 location={location}
-                onLinkClick={closeSidebar}
```
Comment (fk) [inline www/src/components/sidebar/sidebar.js]: Ah, `closeSidebar`! That of course made a lot more sense in the context of Gatsby v1. All the sidebar stuff started as #6076 against `www` on v1 (and mostly borrowing the reactjs.org sidebar implementation 🙏), where the sidebars were part of the `layout`, and never got removed from the DOM when switching `location`. 

IMO `closeSidebar` does still make sense for Gatsby v2 `www` for mobile and the anchor links in the Tutorial (https://next.gatsbyjs.org/tutorial/part-zero/#overview-of-core-technologies)…having a hard time putting thoughts into words, long day…lemme record a video! 😄 

OK, here's the one good thing `closeSidebar` is still doing and IMO is worth preserving:
https://www.dropbox.com/s/yj6dl1wx788vpwm/closeSidebar.mov?dl=0

**Comment by LekoArts** [comment]
@kkemple @fk What's the status here? 😊

**Comment by fk** [comment]
@LekoArts 🤗
Good question…"forgotten" is the status I guess! 😅 

IIRC my comment explains that while this PR does fix the connected issue, it does so at the expense of `closeSidebar`. IMO we should keep closing the sidebar when clicking a link.

**Comment by LekoArts** [comment]
@fk So could we close this PR then?

**Comment by calcsam** [comment]
Let's close this as it's stale and merge conflicts.

@fk you can re-open if we want to do something like this later.

**Pull request closed without merging**